thesis-gui/src/components/my-account/MyAccount.tsx

```diff
@@ -18,45 +23,71 @@ const UserInfo: React.FC = () => {
                 </IonAvatar>
                 <IonCardContent>
                     <IonCardTitle>
-                        Nickname
+                        { user!.nick }
                     </IonCardTitle>
                     <IonCardSubtitle>
-                        nick@example.com
+                        { user!.email }
                     </IonCardSubtitle>
                 </IonCardContent>
-                <IonButton expand="block">
+                <IonButton expand="block" onClick={ signOut }>
                     Sign Out
                 </IonButton>
             </IonCardContent>
         </IonCard>
     );
 }
 
+const CategoryRanksItem: React.FC<{ name: string, rank: number }> = ({ name, rank }) => {
+    return (
+        <IonItem>
+            <IonLabel>{ name }</IonLabel>
+            <IonLabel slot="end">
+                { rank }
+            </IonLabel>
+        </IonItem>
+    );
+}
+
 const Ranking = () => {
-  return (
+    const { user } = useUserContext();
+    const { categories } = useCategoryContext();
+
+    const getCategoryNameWithId = (id: number) => {
+        for (let category of categories) {
+            if (category.id === id) {
+                return category.name;
+            }
+        }
+        return '';
+    }
+
+    const categoryRanksItems = Object.entries(user!.categoryRanks).filter(([_, rank]) => rank > 0).map(([id, rank]) => {
```

Comment:
Currently every user start with rank = 0.5. Is this `filter `necessary?

Reply:
Is it possible that the rank will go down to 0 after some time playing?

Reply:
No, it should not be possible. It can reach **nearly** 0, but not the 0.0